### PR TITLE
[CI] Expanded RosettaIntegrationTests dirtyWhen attribute [Berk]

### LINKED ISSUE #13656: Rosetta integration tests should run on archive / node changes
Rosetta integration tests only run when `src/app/rosetta/` or the job definition is dirty. Node / archive changes could break Rosetta integration, so the integration test should run on those cases too.

### PR DESCRIPTION
**Pull request by dkijania**
Explain your changes:
Expanded dirty when attribute for rosetta integration tests as per request. Now, tests are run on archive an node changes.

Explain how you tested your changes:

Run CI

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #13656

### CODE REVIEW

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Review by joaosreis — APPROVED** [review]
LGTM, those should be the necessary conditions

File: buildkite/src/Jobs/Test/RosettaIntegrationTests.dhall [hunk 1]
```diff
@@ -13,9 +13,14 @@ let Size = ../../Command/Size.dhall
...
 let DockerImage = ../../Command/DockerImage.dhall
 let DebianVersions = ../../Constants/DebianVersions.dhall
+
 let dirtyWhen = [ 
-  S.strictlyStart (S.contains "src/app/rosetta")
-  , S.strictlyStart (S.contains "buildkite/src/Jobs/Test/RosettaIntegrationTests")
+  S.strictlyStart (S.contains "src/app/rosetta"),
+  S.strictlyStart (S.contains "src/lib"),
+  S.strictlyStart (S.contains "src/app/archive"),
+  S.strictlyStart (S.contains "src/app/cli"),
```
Comment (deepthiskumar) [inline buildkite/src/Jobs/Test/RosettaIntegrationTests.dhall:21]: Could you confirm if this is needed and if not, remove?

Comment (joaosreis) [reply buildkite/src/Jobs/Test/RosettaIntegrationTests.dhall:21]: These should be needed to catch changes in the archive / node that could break Rosetta implementation.

Comment (dkijania) [reply buildkite/src/Jobs/Test/RosettaIntegrationTests.dhall:21]: @deepthiskumar yes i might carry away with my `dirtyWhen` expansion. `src/app/cli` looks like it is not needed

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
Fixes #13665

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by bkase** [comment]
!approved-for-mainnet

**Pull request merged**